src/v/pandaproxy/rest/proxy.cc

```diff
@@ -41,9 +42,20 @@ auto wrap(Handler h) {
             // Will throw 400 & 401 if auth fails
             auto auth_result = rq.service().authenticator().authenticate(
               *rq.req);
-            // Will throw 403 if user enabled HTTP Basic Auth but
-            // did not give the authorization header.
-            auth_result.require_authenticated();
+            try {
+                auth_result.require_authenticated();
+            } catch (const ss::httpd::base_exception& ex) {
+                // The failure from require_authenticated() will throw 401
+                // instead of 403. So convert the exception to 403.
```

Comment:
Is this the same behaviour as the other implementations?

Reply:
I'm dropping this commit since it may not be necessary. I'll look into error compatibility in another PR.